File: 2024_04/0411/beginner/beginner.java
import java.util.Scanner;
import java.util.ArrayList;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        int M = scanner.nextInt();
        ArrayList<Integer> li = new ArrayList<>();

        for (int i = 0; i < M; ++i) {
            li.add(scanner.nextInt());
        }

        int res = 0;
        for (int i = 1; i <= N; ++i) {
            for (int j = 0; j < M; ++j) {
                if (i % li.get(j) == 0) {
                    res += i;
                    break;
                }
            }
        }

        System.out.println(res);
        scanner.close();
    }
}

File: 2024_04/0411/beginner/beginner.c
#include <stdio.h>

int main() {
    int N, M;
    scanf("%d %d", &N, &M);
    int li[M];
    int i;
    for (i = 0; i < M; ++i) {
        scanf("%d", &li[i]);
    }
    int res = 0;
    int j;
    for (i = 1; i <= N; ++i) {
        for (j = 0; j < M; ++j) {
            if (i % li[j] == 0) {
                res += i;
                break;
            }
        }
    }
    printf("%d\n", res);
    return 0;
}
